### PR TITLE
Documented difference of this cmdlet in Unix v Window

### PR DESCRIPTION
**Pull request by 1RedOne**
Summarized the information here in the notes for this cmdlet.

https://github.com/PowerShell/PowerShell/issues/4521#issuecomment-324931106

Version(s) of document impacted
------------------------------
- [x] Impacts 6.next document
- [x] Impacts 6 document
- [ ] Impacts 5.1 document
- [ ] Impacts 5.0 document
- [ ] Impacts 4.0 document
- [ ] Impacts 3.0 document

Reason(s) for not updating all version of documents
--------------------------------------------------
- [x] The documented feature was introduced in version (list version here) of PowerShell
- [ ] This issue only shows up in version (list version(s) here) of the document
- [ ] This PR partially fixes the issue, and issue #<insert here> tracks the remaining work

### CODE REVIEW

**Comment by zjalexander** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[reference/6/Microsoft.PowerShell.Management/Start-Process.md](https://github.com/1RedOne/PowerShell-Docs/blob/patch-1/reference/6/Microsoft.PowerShell.Management/Start-Process.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.management/start-process?view=powershell-6&branch=pr-en-us-2989) (powershell-6) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C10%5C4%5Cbab27324-4c59-a348-1473-9c6d587e4020%5CPullRequest%5C201810041355132783-2989%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=E2vKCzpHOp%2BLRNF6nBs24YDxM7OrlkG5NmtaSYVKYpU%3D&st=2018-10-04T14%3A04%3A39Z&se=2018-11-04T14%3A09%3A39Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: reference/6/Microsoft.PowerShell.Management/Start-Process.md [hunk 1]
```diff
@@ -396,6 +396,8 @@ Otherwise, this cmdlet does not return any output.
 ## NOTES
 * This cmdlet is implemented by using the **Start** method of the **System.Diagnostics.Process** class. For more information about this method, see [Process.Start Method](https://msdn.microsoft.com/library/system.diagnostics.process.start) in the MSDN library.
 
+* When using PowerShell Core on Linux, to open a new process within a new window (similar to the default behavior when using Start-Process in Windows), run the cmdlet with the `-UseNewEnvironment -Wait` parameters.  This is done to prevent the new process from blocking PowerShell's control of keyboard input.
```
Comment (mklement0) [inline reference/6/Microsoft.PowerShell.Management/Start-Process.md]: Thanks for taking this on, @1RedOne, but this information is not correct:

* Creating a new _window_ is fundamentally unsupported on Unix-like platforms (macOS and Linux).
   * As an aside: Please either mention both macOS and Linux explicitly, if both platforms are affected or use a term such as _Unix-like platforms_; @sdwheeler, is there established terminology?

* `-Wait` is only needed if you're launching a terminal-based (non-GUI) utility (to prevent the keyboard / output issue), as opposed to asynchronously launching a GUI, such as `gedit`, which is the other legitimate use case.

   * However, the only reason to run a terminal-based utility with `Start-Process -Wait` is in the _unusual_ event that you want to ignore the current process' environment and start with a fresh one via `-UseNewEnvironment` - without that, the utility can just be invoked _directly_ (which makes the invocation _implicitly_ synchronous and also keeps the new process' standard streams connected to PowerShell's).

In short, on Unix-like platforms:

* `-NoNewWindow` is _invariably implied_ - you'll never get a new terminal window.
* Use `Start-Process` _only_ in these scenarios:
   * To launch a GUI application that has a blocking CLI (e.g., `gedit`, `firefox`) asynchronously - typical only on Linux.
   * To launch a terminal-based utility with `-UseNewEnvironment` to start with a fresh environment, which then also requires use of `-Wait` to prevent the keyboard-control issues and/or preventing output from arriving asynchronously. With `-Wait`, output will print synchronously to the _terminal_, but due to the process not being connected to PowerShell's streams you won't able to capture or suppress it - which is the price of using `Start-Process`.
